### PR TITLE
implement crc

### PR DESCRIPTION
**Pull request by Lotterleben**
closes https://github.com/ferrous-systems/nrfdfu-rs/issues/8

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -210,6 +213,26 @@ impl BootloaderConnection {
         Ok(())
     }
 
+    fn check_crc(&self, data: &[u8], received_crc: u32, initial: u32) -> Result<u32> {
+
+        let mut digest = crc32::Digest::new_with_initial(crc32::IEEE, initial);
+        digest.write(data);
+        let expected_crc = digest.sum32() & 0xffffffff;
+
+        match expected_crc == received_crc {
```
Comment (jonas-schievink) [inline src/main.rs:222]: wouldn't `if` be simpler here?

**Review by jonas-schievink — APPROVED** [review]
Looks good, and seems to work fine. Feel free to r+

File: src/main.rs (hunk 1 above)
Comment (Lotterleben) [reply src/main.rs:222]: I got a little `match`-excited and wasn't sure what the most idiomatic choice would be :D

**Pull request merged**